### PR TITLE
[PLAT-12233] Error correlation browser testing

### PR DESCRIPTION
**Pull request by gingerbenw**
## Goal

Add end to end tests for error correlation in browser

### CODE REVIEW

File: test/browser/features/fixtures/packages/error-correlation/package.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+    "name": "error-correlation",
+    "private": true,
+    "scripts": {
+        "build": "rollup --config ../rollup.config.mjs",
+        "clean": "rm -rf dist"
+    },
+    "dependencies": {
+        "@bugsnag/browser": "^7.25.0-alpha.0"
```
Comment (yousif-bugsnag) [inline test/browser/features/fixtures/packages/error-correlation/package.json:9]: is there a particular reason for using the `@bugsnag/browser` dependency directly here and not `@bugsnag/js`?

Comment (gingerbenw) [reply test/browser/features/fixtures/packages/error-correlation/package.json:9]: only that it's in the `browser` fixture, and not node, meaning for a smaller install

Comment (yousif-bugsnag) [reply test/browser/features/fixtures/packages/error-correlation/package.json:9]: ah okay, makes sense 👍🏾

File: test/browser/features/error-correlation.feature [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    # Assert on the span, and store the trace id and span id
+    Then the trace payload field "resourceSpans.0.scopeSpans.0.spans.0.id" is stored as the value "spanId"
+    And the trace payload field "resourceSpans.0.scopeSpans.0.spans.0.traceid" is stored as the value "traceId"
+
+    # Check span id is included in the error
+    And the error payload field "events.0.correlation.spanId" equals the stored value "spanId"
+    And the error payload field "events.0.correlation.traceId" equals the stored value "traceId"
+
+  Scenario: Reported errors include the parent trace and span id
```
Comment (yousif-bugsnag) [inline test/browser/features/error-correlation.feature:25]: I found this scenario a bit confusing as it reads like the error should always include the parent trace and span id when there's a nested span.

Maybe something like 'Reported errors do not include the trace and span id if the span is not the current context' - or maybe even roll them both into one scenario?

Also might be worth adding a test for when there are no open spans?

File: test/browser/features/error-correlation.feature [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+    # Assert on the span, and store the trace id and span id
+    Then the trace payload field "resourceSpans.0.scopeSpans.0.spans.0.id" is stored as the value "spanId"
+    And the trace payload field "resourceSpans.0.scopeSpans.0.spans.0.traceid" is stored as the value "traceId"
+
+    # Check span id is included in the error
+    And the error payload field "events.0.correlation.spanId" equals the stored value "spanId"
+    And the error payload field "events.0.correlation.traceId" equals the stored value "traceId"
+
+  Scenario: Reported errors do not include the trace and span id if the span is not the current context
```
Comment (gingerbenw) [inline test/browser/features/error-correlation.feature:33]: @yousif-bugsnag  does this make a bit more sense? and am I missing something simple here? it's failing on like 62 at the moment

Comment (yousif-bugsnag) [reply test/browser/features/error-correlation.feature:33]: yeah that definitely makes more sense 👍🏾 

not sure why that is failing though as it seems sensible enough 🤔 would need to take a look at the traces to debug it

**Review by lemnik — APPROVED** [review]
LGTM

**Pull request merged**